test/controllers/works_controller_test.rb

```diff
@@ -34,25 +34,58 @@
   INVALID_CATEGORIES = ["nope", "42", "", "  ", "albumstrailingtext"]
 
   describe "index" do
-    it "succeeds when there are works" do
-      get works_path
+    before do
+      @user = users(:grace)
+    end
+    describe "logged in users" do
+      it "succeeds when there are works" do
+        perform_login(@user)
+        get works_path
 
-      must_respond_with :success
+        must_respond_with :success
+      end
+
+      it "succeeds when there are no works" do
+        perform_login(@user)
+        Work.all do |work|
+          work.destroy
+        end
+
+        get works_path
+
+        must_respond_with :success
+      end
     end
+    describe "logged out users" do
+      before do
+        perform_login(@user)
+        post logout_path
+      end
+      it "redirects when there are works" do
+        get works_path
 
-    it "succeeds when there are no works" do
-      Work.all do |work|
-        work.destroy
+        expect(flash[:error]).must_equal "You must be logged in to do this action"
+        must_respond_with :redirect
+        must_redirect_to root_path
       end
 
-      get works_path
+      it "redirects when there are no works" do
+        Work.all do |work|
+          work.destroy
+        end
 
-      must_respond_with :success
+        get works_path
+
+        expect(flash[:error]).must_equal "You must be logged in to do this action"
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
     end
   end
 
   describe "new" do
-    it "succeeds" do
+    it "succeeds when user is logged in" do
```

Comment:
I notice you prevent the guest user from doing this, but don't have a test for it.